dateparser/date.py

```diff
@@ -314,16 +309,8 @@ def __init__(self, languages=None, locales=None, region=None, try_previous_local
             raise TypeError("try_previous_locales argument must be a boolean (%r given)"
                             % type(try_previous_locales))
 
-        if not isinstance(use_given_order, bool):
-            raise TypeError("use_given_order argument must be a boolean (%r given)"
-                            % type(use_given_order))
-
-        if not locales and use_given_order:
-            raise ValueError("locales must be given if use_given_order is True")
-
```

Comment:
Based on the conversation so far, [my vote is to keep these](https://github.com/scrapinghub/dateparser/pull/789#discussion_r492014507).

Reply:
Hey @Gallaecio if this is the way to go the I believe https://github.com/scrapinghub/dateparser/pull/789/commits/94c96fd7585a2b00bbe1ef3670373f673eb82ec7 should cover it.

Let me know what you think